config/v1/types.go

```diff
@@ -80,3 +81,134 @@ type LeaderElection struct {
 	// leader election is enabled.
 	RetryPeriod metav1.Duration `json:"retryPeriod,omitempty" protobuf:"bytes,6,opt,name=retryPeriod"`
 }
+
+// StringSource allows specifying a string inline, or externally via env var or file.
+// When it contains only a string value, it marshals to a simple JSON string.
+type StringSource struct {
+	// StringSourceSpec specifies the string value, or external location
+	StringSourceSpec `json:",inline" protobuf:"bytes,1,opt,name=stringSourceSpec"`
+}
+
+// StringSourceSpec specifies a string value, or external location
+type StringSourceSpec struct {
+	// Value specifies the cleartext value, or an encrypted value if keyFile is specified.
+	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
+
+	// Env specifies an envvar containing the cleartext value, or an encrypted value if the keyFile is specified.
+	Env string `json:"env" protobuf:"bytes,2,opt,name=env"`
+
+	// File references a file containing the cleartext value, or an encrypted value if a keyFile is specified.
+	File string `json:"file" protobuf:"bytes,3,opt,name=file"`
+
+	// KeyFile references a file containing the key to use to decrypt the value.
+	KeyFile string `json:"keyFile" protobuf:"bytes,4,opt,name=keyFile"`
+}
+
+// RemoteConnectionInfo holds information necessary for establishing a remote connection
+type RemoteConnectionInfo struct {
+	// URL is the remote URL to connect to
+	URL string `json:"url" protobuf:"bytes,1,opt,name=url"`
+	// CA is the CA for verifying TLS connections
+	CA string `json:"ca" protobuf:"bytes,2,opt,name=ca"`
+	// CertInfo is the TLS client cert information to present
+	// this is anonymous so that we can inline it for serialization
+	CertInfo `json:",inline" protobuf:"bytes,3,opt,name=certInfo"`
+}
+
+// AdmissionPluginConfig holds the necessary configuration options for admission plugins
+type AdmissionPluginConfig struct {
+	// Location is the path to a configuration file that contains the plugin's
+	// configuration
+	Location string `json:"location" protobuf:"bytes,1,opt,name=location"`
```

Comment:
nit: ConfigurationPath ?

Reply:
or even 'ConfigurationFilePath` to match the admission

Reply:
Potentiall later.  This is compatible with what existed before for now.